app/assets/src/components/PipelineSampleReport.jsx

```diff
@@ -100,6 +100,7 @@ class PipelineSampleReport extends React.Component {
     // Starting state is default values which are to be set later.
     this.state = {
       taxonomy_details: [],
+      topScoringTaxa: [],
```

Comment:
Should the top scoring come from server? Could you have do this on client side?

Reply:
I think it should. We're trying to serve resource-limited areas so shouldn't do too much data processing in the browser. Semantically the logic also fits well in that place of the code.

Reply:
The most scarce resource is actually network :)
But it is ok to do it server side.